frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useRef, useState } from "react";
+import { t } from "ttag";
+import { isNotNull } from "metabase/core/utils/types";
+import Button from "metabase/core/components/Button";
+import Input from "metabase/core/components/Input/Input";
+import Tooltip from "metabase/core/components/Tooltip";
+import MetabaseSettings from "metabase/lib/settings";
+import type { Expression } from "metabase-types/types/Query";
+import { isExpression } from "metabase-lib/expressions";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import ExpressionEditorTextfield from "./ExpressionEditorTextfield";
+import {
+  ActionButtonsWrapper,
+  Container,
+  Divider,
+  ExpressionFieldWrapper,
+  FieldTitle,
+  FieldWrapper,
+  Footer,
+  InfoLink,
+  RemoveLink,
+  StyledFieldTitleIcon,
+} from "./ExpressionWidget.styled";
+
+const EXPRESSIONS_DOCUMENTATION_URL = MetabaseSettings.docsUrl(
+  "questions/query-builder/expressions",
+);
+
+export interface ExpressionWidgetProps {
+  query: StructuredQuery;
+  expression: Expression | undefined;
+  name: string | undefined;
+
+  reportTimezone: string;
+
+  onChangeExpression: (name: string, expression: Expression) => void;
+  onRemoveExpression?: (name: string) => void;
+  onClose?: () => void;
+}
+
+const ExpressionWidget = (props: ExpressionWidgetProps): JSX.Element => {
+  const {
+    query,
+    name: initialName,
+    expression: initialExpression,
+    reportTimezone,
+    onChangeExpression,
+    onRemoveExpression,
+    onClose,
+  } = props;
+
+  const [name, setName] = useState(initialName || "");
+  const [expression, setExpression] = useState<Expression | null>(
+    initialExpression || null,
+  );
+  const [error, setError] = useState<string | null>(null);
+
+  const helpTextTargetRef = useRef(null);
+
+  const isValid = !!name && !error && isExpression(expression);
+
+  const handleCommit = () => {
+    if (isValid && isNotNull(expression)) {
+      onChangeExpression(name, expression);
+      onClose && onClose();
+    }
+  };
+
+  return (
+    <Container>
+      <ExpressionFieldWrapper>
+        <FieldTitle>
+          {t`Expression`}
+
+          <Tooltip
+            tooltip={t`You can reference columns here in functions or equations, like: floor([Price] - [Discount]). Click for documentation.`}
+            placement="right"
+            maxWidth={332}
+          >
+            <InfoLink
+              target="_blank"
+              href={EXPRESSIONS_DOCUMENTATION_URL}
+              data-testid="expression-widget-info-link"
```

Comment:
I believe this is from https://github.com/metabase/metabase/pull/29113. I've checked and seems nobody's using this anymore, so it should be removed.